internal/cephfs/core/quiesce.go

```diff
@@ -0,0 +1,216 @@
+}
+
+type Volume struct {
+	VolumeID  string
+	ClusterID string
+}
+
+type fsQuiesce struct {
+	fsName  string
+	volumes []Volume
+	// subVolumeGroupMapping is a map of subvolumes to groups.
+	subVolumeGroupMapping map[string][]string
+	fsa                   *admin.FSAdmin
+}
+
+// NewFSQuiesce returns a new instance of fsQuiesce. It
+// take the filesystem name, the list of volumes to be quiesced, the mapping of
+// subvolumes to groups and the cluster connection as input.
+func NewFSQuiesce(
+	fsName string,
+	volumes []Volume,
+	mapping map[string][]string,
+	conn *util.ClusterConnection,
+) (FSQuiesceClient, error) {
+	fsa, err := conn.GetFSAdmin()
+	if err != nil {
+		return nil, err
+	}
+
+	return &fsQuiesce{
+		fsName:                fsName,
+		volumes:               volumes,
+		subVolumeGroupMapping: mapping,
+		fsa:                   fsa,
+	}, nil
+}
+
+// GetVolumes returns the list of volumes in the filesystem that are to be
+// quiesced.
+func (fq *fsQuiesce) GetVolumes() []Volume {
+	return fq.volumes
+}
+
+// getMembers returns the list of names in the format
+// group/subvolume that are to be quiesced. This is the format that the
+// ceph fs quiesce expects.
+// Example: ["group1/subvolume1", "group1/subvolume2", "group2/subvolume1"].
+func (fq *fsQuiesce) getMembers() []string {
+	volName := []string{}
+	for svg, sb := range fq.subVolumeGroupMapping {
+		for _, s := range sb {
+			// name is in the format of subvolume/group
+			// TODO: Remove this hardcoding as it is not required when we get next release of ceph
+			name := svg + "/" + s + "?q=30"
+			volName = append(volName, name)
+		}
+	}
+
+	return volName
+}
+
+func (fq *fsQuiesce) FSQuiesce(
+	ctx context.Context,
+	reserveName string,
+) (*admin.QuiesceInfo, error) {
+	opt := &admin.FSQuiesceOptions{
+		Timeout:    180,
+		AwaitFor:   0,
+		Expiration: 180,
+	}
+	resp, err := fq.fsa.FSQuiesce(fq.fsName, admin.NoGroup, fq.getMembers(), reserveName, opt)
+	if err != nil {
+		log.ErrorLog(ctx, "failed to quiesce filesystem %s", err)
+
+		return nil, err
+	}
+	qInfo := resp.Sets[reserveName]
+
+	return &qInfo, nil
+}
+
+func (fq *fsQuiesce) FSQuiesceWithExpireTimeout(ctx context.Context,
+	reserveName string,
+) (*admin.QuiesceInfo, error) {
+	opt := &admin.FSQuiesceOptions{
+		Timeout:    180,
+		AwaitFor:   0,
+		Expiration: 180,
+	}
+	resp, err := fq.fsa.FSQuiesce(fq.fsName, admin.NoGroup, fq.getMembers(), reserveName, opt)
+	if err != nil {
+		log.ErrorLog(ctx, "failed to quiesce filesystem with expire timeout %s", err)
+
+		return nil, err
+	}
+
+	qInfo := resp.Sets[reserveName]
+
+	return &qInfo, nil
+}
+
+func (fq *fsQuiesce) ResetFSQuiesce(ctx context.Context,
+	reserveName string,
+) (*admin.QuiesceInfo, error) {
+	opt := &admin.FSQuiesceOptions{
+		Reset:      true,
+		AwaitFor:   0,
+		Timeout:    180,
+		Expiration: 180,
+	}
+	resp, err := fq.fsa.FSQuiesce(fq.fsName, admin.NoGroup, fq.getMembers(), reserveName, opt)
+	if err != nil {
```

Comment:
Please add some comments regarding what reset does. 
Do we need timeout and expiration here too?

Reply:
Yes it takes the timeout value and added comment what it does